db/migrate/20240305082008_link_filters_to_cached_aggregation.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+class LinkFiltersToCachedAggregation < ActiveRecord::Migration[7.0]
+  class CachedAggregation < ApplicationRecord
+    belongs_to :group, optional: true
+    belongs_to :charge
+  end
+
+  class Fee < ApplicationRecord
+    belongs_to :group, -> { with_discarded }, optional: true
+    belongs_to :charge, -> { with_discarded }, optional: true
+    belongs_to :charge_filter, -> { with_discarded }, optional: true
+  end
+
+  class AdjustedFee < ApplicationRecord
+    belongs_to :charge, optional: true
+    belongs_to :group, optional: true
+  end
+
+  class QuantifiedEvent < ApplicationRecord
+    belongs_to :group, optional: true
+  end
+
+  def up
+    # NOTE: Associate cached aggregations with charge filters
+    CachedAggregation.where.associated(:group).where(charge_filter_id: nil).find_each do |agg|
+      link_charge_filter(agg)
+    end
+
+    # NOTE: Associate fees with charge filters
+    Fee.where.associated(:group).where(charge_filter_id: nil).find_each do |fee|
+      link_charge_filter(fee)
+    end
+
+    # NOTE: Associate adjusted fees with charge filters
+    AdjustedFee.where.associated(:group).where(charge_filter_id: nil).find_each do |fee|
+      link_charge_filter(fee)
+    end
+  end
+
+  def down
+    CachedAggregation.where.associated(:group).update_all(charge_filter_id: nil)
+    Fee.where.associated(:group).update_all(charge_filter_id: nil)
```

Comment:
Rails/SkipsModelValidations: Avoid using `update_all` because it skips validations. (https://guides.rubyonrails.org/active_record_validations.html#skipping-validations)